lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -891,30 +906,48 @@ Value MemOp::getPortNamed(StringAttr name) {
 //===----------------------------------------------------------------------===//
 
 static LogicalResult verifyConnectOp(ConnectOp connect) {
-  FIRRTLType destType =
-      connect.dest().getType().cast<FIRRTLType>().getPassiveType();
-  FIRRTLType srcType =
-      connect.src().getType().cast<FIRRTLType>().getPassiveType();
+  FIRRTLType destType = connect.dest().getType().cast<FIRRTLType>();
+  FIRRTLType srcType = connect.src().getType().cast<FIRRTLType>();
 
   // Analog types cannot be connected and must be attached.
   if (destType.isa<AnalogType>() || srcType.isa<AnalogType>())
     return connect.emitError("analog types may not be connected");
 
+  if (destType.getPassiveType().isa<BundleType>()) {
```

Comment:
Anyone think it would be a bad idea to implement a helper for this that doesn't potentially intern a new type?

Reply:
```c++
bool isBundleType(Operation * op) {
  if(op.isa<FlipType>()) {
    return op.cast<FlipType>().getElement().isa<BundleType>();
  return op.isa<BundleType>();
}
```

Reply:
Seems reasonable.

Reply:
Added isBundleType